Changes for opting long url or null on link creation  error

sojanpr opened the pull request:
additional APIs added with `defaultToLongUrl`  for creating links sync / async

@aaustin @EvangelosG

chandsie commented on Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java:
```diff
@@ -501,15 +501,56 @@ public void registerView(@Nullable RegisterViewStatusListener callback) {
 
+    /**
+     * Creates a short url for the BUO synchronously.
+     * xa
```
looks like an accidental comment

chandsie commented on Branch-SDK/src/io/branch/referral/BranchShortLinkBuilder.java:
```diff
@@ -127,7 +127,19 @@ public BranchShortLinkBuilder setType(int type) {
+     * <p>Configures and requests a short URL to be generated by the Branch servers, via a synchronous
+     * call; with a duration specified within which an app session should be matched to the link.</p>
+     *
+     * @param defaultToLongUrl A {@link Boolean} whose value should be false if a  long url is not intended to return on failure.
```
a bit nitpicky: the argument isn't a Boolean, but a primitive boolean (lowercase b)

chandsie replied:
Also nitpicky, there's an extra space between "false if a" and "long url"

chandsie commented on Branch-SDK/src/io/branch/referral/BranchShortLinkBuilder.java:
```diff
@@ -137,10 +149,23 @@ public String getShortUrl() {
+     * @param defaultToLongUrl A {@link Boolean} whose value should be false if a  long url is not intended to return on failure.
+     *                         Setting this it false will return NULL for url on failure else a long url.
+     */
+    public void generateShortUrl(Branch.BranchLinkCreateListener callback, Boolean defaultToLongUrl) {
```
I believe the `defaultToLongUrl` argument can be just a primitive boolean here. This overhead is probably negligible but consistency would be nice.

chandsie commented on Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java:
```diff
@@ -501,15 +501,56 @@ public void registerView(@Nullable RegisterViewStatusListener callback) {
+     *
+     * @param context          {@link Context} instance
+     * @param linkProperties   An object of {@link LinkProperties} specifying the properties of this link
+     * @param defaultToLongUrl A {@link Boolean} specifies if a long url should be returned in case of link creation error
```
a bit nitpicky: the argument isn't a Boolean, but a primitive boolean (lowercase b).
Mentioned elsewhere as well

chandsie commented on Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java:
```diff
@@ -501,15 +501,56 @@ public void registerView(@Nullable RegisterViewStatusListener callback) {
+     * @param context          {@link Context} instance
+     * @param linkProperties   An object of {@link LinkProperties} specifying the properties of this link
+     * @param defaultToLongUrl A {@link Boolean} specifies if a long url should be returned in case of link creation error
+     *                         If set to false NULL is returned in case of link creation error
```
nitpicky again (sorry!), but need a comma between false and NULL.

chandsie commented on Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java:
```diff
@@ -501,15 +501,56 @@ public void registerView(@Nullable RegisterViewStatusListener callback) {
+     * @param context          {@link Context} instance
+     * @param linkProperties   An object of {@link LinkProperties} specifying the properties of this link
+     * @param callback         An instance of {@link io.branch.referral.Branch.BranchLinkCreateListener} to receive the results
+     * @param defaultToLongUrl A {@link Boolean} specifies if a long url should be returned in case of link creation error
```
argument is boolean, not Boolean

chandsie commented on Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java:
```diff
@@ -501,15 +501,56 @@ public void registerView(@Nullable RegisterViewStatusListener callback) {
+     * @param linkProperties   An object of {@link LinkProperties} specifying the properties of this link
+     * @param callback         An instance of {@link io.branch.referral.Branch.BranchLinkCreateListener} to receive the results
+     * @param defaultToLongUrl A {@link Boolean} specifies if a long url should be returned in case of link creation error
+     *                         If set to false NULL is returned in case of link creation error
```
same comma nitpick as before

sojanpr commented:
@chandsie  Thank you for the great review. Everything in the review comments makes perfect sense and we have added fix for it. We have primitive type vs Object issue in quite a few places in documentation. Addressed only for the files in this PR now but will be taken care for other files soon.

Thank you for your efforts to make Branch SDK better.

aaustin commented:
👍
